crates/runtime/src/cluster/datafusion/codec/spice_physical_codec.rs

```diff
@@ -0,0 +1,105 @@
+use crate::Runtime;
+use crate::metrics::telemetry::track_bytes_processed;
+use arrow_schema::Schema;
+use ballista_core::serde::BallistaPhysicalExtensionCodec;
+use datafusion::common::{DataFusionError, Result, exec_err};
+use datafusion::physical_plan::ExecutionPlan;
+use datafusion_expr::ScalarUDF;
+use datafusion_expr::registry::FunctionRegistry;
+use datafusion_proto::generated::datafusion_common;
+use datafusion_proto::physical_plan::PhysicalExtensionCodec;
+use prost::Message;
+use runtime_datafusion::execution_plan::schema_cast::SchemaCastScanExec;
+use runtime_datafusion::extension::bytes_processed::BytesProcessedExec;
+use runtime_proto::{BytesProcessedExecNode, SchemaCastScanExecNode};
+use std::fmt::Debug;
+use std::sync::Arc;
+
```

Comment:
Simplified with new protos. Also in DF50+, our extension is only called if there is no match on a DataFusion native node (so AFAICT we don't even have to fall back, although, doing it for sanity).